simpledotorg/simple-server
# Progress Tab: Phase 2 Monthly Reports, HTN Reports, Overdue Call Result

**Story cards:** 
[sc-7633](https://app.shortcut.com/simpledotorg/story/7633/phase-2-add-total-overdue-patients-contacted-to-achievements-card-in-progress-tab-home-page)
[9294](https://app.shortcut.com/simpledotorg/story/9294/phase-2-populate-monthly-reports)
[7365](https://app.shortcut.com/simpledotorg/story/7365/phase-2-populate-htn-reports)

## Because

- Monthly Reports 
  - Current monthly reports don't show monthly data breakdown by diagnosis and gender on one page
- The achievement section does not have over due call results section

## This addresses

- Monthly Reports 
  - Show monthly data breakdown by diagnosis and gender on one page
- Adds the number of total overdue calls made under the Achievement section
- [refactor]Hypertension Reports
  - Extract the partial into a view component

## Test instructions

- Ensure to turn on the feature flag -`new_progress_tab_v2`
- Goto Reports -> <region> -> Progress
- Check 

1. There should be an overdue calls section under Achievements 
<img width="252" alt="Screenshot 2022-11-02 at 6 41 00 PM" src="https://user-images.githubusercontent.com/32141642/199498284-e42e453f-5c9c-4e8f-988c-ac551fd3c531.png">

2. Monthly reports should have a breakdown by diagnosis and gender on the same page
<img width="252" alt="Screenshot 2022-11-02 at 6 40 43 PM" src="https://user-images.githubusercontent.com/32141642/199498305-c8558759-7486-415d-9409-ba6fd14bc861.png">
<img width="252" alt="Screenshot 2022-11-02 at 6 40 48 PM" src="https://user-images.githubusercontent.com/32141642/199498312-9026c9f0-39bc-4d7d-ac38-5e393c564828.png">

3. Hypertension reports should have all the data as in the previous phase
<img width="252" alt="Screenshot 2022-11-02 at 6 41 07 PM" src="https://user-images.githubusercontent.com/32141642/199498331-b4c7b2ef-7866-4890-9ee8-c103ee23574e.png">
<img width="252" alt="Screenshot 2022-11-02 at 6 41 14 PM" src="https://user-images.githubusercontent.com/32141642/199498343-0de89f2e-cd3f-4f46-a7ea-769cee61393b.png">
<img width="252" alt="Screenshot 2022-11-02 at 6 41 19 PM" src="https://user-images.githubusercontent.com/32141642/199498347-15002823-c432-4431-ab99-3c68dfaab305.png">
<img width="252" alt="Screenshot 2022-11-02 at 6 41 29 PM" src="https://user-images.githubusercontent.com/32141642/199498352-576effa3-1ad7-4be2-b959-913dea74d838.png">
